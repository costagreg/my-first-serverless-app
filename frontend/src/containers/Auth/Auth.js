import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import LoginForm from '../../components/LoginForm'
import SignupForm from '../../components/SignupForm'
import AuthButtons from '../../components/AuthButtons'
import { signup, signupError } from '../../actions/signup'

const mapToProps = (state) => ({
  error: state.signup.error,
})

export const Auth = ({ signup, signupError, error }) => {
  const [showLogin, setShowLogin] = useState(true)

  return (
    <>
      <AuthButtons setShowLogin={setShowLogin} />
      {showLogin ? (
        <LoginForm />
      ) : (
        <SignupForm signup={signup} signupError={signupError} error={error} />
      )}
    </>
  )
}

Auth.propTypes = {}

export default connect(mapToProps, { signup, signupError })(Auth)
