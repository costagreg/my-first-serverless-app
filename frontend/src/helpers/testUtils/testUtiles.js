import React from 'react'
import thunk from 'redux-thunk'
import { render as rtlRender } from '@testing-library/react'
import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import reducer from '../../reducers'

function render(
  ui,
  {
    initialState = {
      appConfig: {
        apiUrl: 'http://api.test.com'
      }
    },
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'

// override render method
export { render }
