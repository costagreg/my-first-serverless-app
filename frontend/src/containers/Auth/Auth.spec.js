import React from 'react'
import Auth from './Auth'
import { render, fireEvent, waitFor } from '../../helpers/testUtils'

describe('Auth', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('@renders', () => {
    it('renders the <Auth/> without any error', () => {
      const { asFragment } = render(<Auth />)

      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('SignUp', () => {
    describe('password does not match', () => {
      it('redenders an error message', () => {
        const { getByRole, getByPlaceholderText, getByText } = render(<Auth />)

        const signupButton = getByRole('button', { name: 'Signup' })

        fireEvent.click(signupButton)

        const emailInput = getByPlaceholderText('Email')
        const passwordInput = getByPlaceholderText('Password')
        const confirmPasswordInput = getByPlaceholderText('Confirm Password')
        const submitButton = getByRole('button', { name: 'Sign up' })

        fireEvent.change(emailInput, { target: { value: 'test@gmail.com' } })
        fireEvent.change(passwordInput, { target: { value: 'password123*' } })
        fireEvent.change(confirmPasswordInput, {
          target: { value: 'password3*' },
        })

        fireEvent.click(submitButton)

        getByText("Password doesn't match")
      })
    })

    describe('password does match', () => {
      it('calls cognito user api and show a message', () => {
        const { getByRole, getByPlaceholderText, getByText } = render(<Auth />)

        const signupButton = getByRole('button', { name: 'Signup' })

        fireEvent.click(signupButton)

        const emailInput = getByPlaceholderText('Email')
        const passwordInput = getByPlaceholderText('Password')
        const confirmPasswordInput = getByPlaceholderText('Confirm Password')
        const submitButton = getByRole('button', { name: 'Sign up' })

        fireEvent.change(emailInput, { target: { value: 'test@gmail.com' } })
        fireEvent.change(passwordInput, { target: { value: 'password123*' } })
        fireEvent.change(confirmPasswordInput, {
          target: { value: 'password123*' },
        })

        fireEvent.click(submitButton)

        getByText('Welcome back cognitousername')
      })
    })
  })
})
