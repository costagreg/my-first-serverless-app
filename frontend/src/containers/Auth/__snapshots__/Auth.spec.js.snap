// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth @renders renders the <Auth/> without any error 1`] = `
<DocumentFragment>
  <div
    class="authButtons"
  >
    <button
      class="authButtons__login"
    >
      Login
    </button>
    <button
      class="authButtons__signup"
    >
      Signup
    </button>
  </div>
  <div
    class="loginForm"
  >
    <form
      class="loginForm__formContainer"
    >
      <input
        class="loginForm__email"
        placeholder="Email"
        type="email"
      />
      <input
        class="loginForm__password"
        placeholder="Password"
        type="password"
      />
      <div
        class="loginForm__buttons"
      >
        <button
          class="loginForm__submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
